packages/currency/src/currencyManager.ts

```diff
@@ -71,17 +71,21 @@ export class CurrencyManager<TMeta = unknown> implements ICurrencyManager<TMeta>
       return this.fromAddress(currencyIdentifier, network);
     }
 
+    if (network && currencyIdentifier.indexOf(network) === -1) {
+      currencyIdentifier = `${currencyIdentifier}-${network}`;
+    }
+
+    const currencyFromId = this.fromId(currencyIdentifier);
+
+    if (currencyFromId) return currencyFromId;
+
```

Comment:
Main purpose of this PR is to prioritize the method CurrencyManager.fromId before fromSymbol, because it matches the direct id of the currency, it's more specific.